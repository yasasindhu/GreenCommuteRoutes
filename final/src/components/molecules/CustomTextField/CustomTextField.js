import React from "react";
import { TextField, ThemeProvider, InputAdornment } from "@material-ui/core";
import LocationOnIcon from "@material-ui/icons/LocationOn";
import customtheme from "../../theme";


const CustomTextField = ({locationPlaceholder}) => {

  return (
    <ThemeProvider theme={customtheme}>
      <TextField
        placeholder={locationPlaceholder}
        variant="outlined"
        InputProps={{
          startAdornment: (
            <InputAdornment position="start">
              <LocationOnIcon style={{ color: "#9bbdcb" }} />
            </InputAdornment>
          ),
        }}
        size="large"
        data-testid = "cutomtextfield"
      />
    </ThemeProvider>
  );
};

export default CustomTextField;
