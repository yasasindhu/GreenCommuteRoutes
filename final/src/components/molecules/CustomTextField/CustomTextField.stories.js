import React from "react";

import CustomTextField from "./CustomTextField";

export default {
  title: "molecules/Custom text field",
  component: CustomTextField,
};
const Template = (args) => <CustomTextField {...args} />;
export const Primary = Template.bind({});

Primary.args = {
  locationPlaceholder : 'Enter your job location'
}
