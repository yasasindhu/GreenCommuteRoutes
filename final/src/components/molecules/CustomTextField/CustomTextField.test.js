import { render, cleanup } from '@testing-library/react';
import CustomTextField from './CustomTextField';


afterEach(cleanup)

describe('CustomTextField Rendered', () => {
  test('CustomTextField should be rendered', () => {
    const { asFragment} = render(<CustomTextField locationPlaceholder = "Search Jobs"/>);
        expect(asFragment()).toMatchSnapshot();
  });
});