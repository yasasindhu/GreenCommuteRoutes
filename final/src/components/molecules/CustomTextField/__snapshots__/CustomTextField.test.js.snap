// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTextField Rendered CustomTextField should be rendered 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root"
    data-testid="cutomtextfield"
  >
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart"
    >
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionStart"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          style="color: rgb(155, 189, 203);"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
          />
        </svg>
      </div>
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart"
        placeholder="Search Jobs"
        type="text"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
        style="padding-left: 8px;"
      >
        <legend
          class="PrivateNotchedOutline-legend-2"
          style="width: 0.01px;"
        >
          <span>
            â€‹
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;
